core/CronArchive/QueueConsumer.php

```diff
@@ -278,6 +280,8 @@ public function getNextArchivesToProcess()
 
             $this->logger->debug("Processing invalidation: $invalidationDesc.");
 
+            $this->repairInvalidationsIfNeeded($invalidatedArchive);
```

Comment:
@diosmosis any thought about doing this in `CronArchive::launchArchivingFor()` after it finished successfully? Technically we don't need to schedule the higher period if it's not successful and also it could prevent a race condition where the `day` archive takes many hours and by the time it's finished the higher period does not exist anymore if I see this right?

Reply:
>  any thought about doing this in CronArchive::launchArchivingFor() after it finished successfully?

I figured it would be better to do it here, since when we invalidate through ArchiveInvalidator, we always add all higher periods. So at the start, we always have the whole list of invalidations.

> Technically we don't need to schedule the higher period if it's not successful and also it could prevent a race condition where the day archive takes many hours and by the time it's finished the higher period does not exist anymore if I see this right?

The only case where the higher period would not exist anymore is if another archiver was processing the same site and picked it up, but this could happen w/o this repairing logic... In that case we'd have to do the intersecting period logic w/ a query on archive_invalidations.

Reply:
>  So at the start, we always have the whole list of invalidations.

you mean then we can batch insert it? I reckon ideally we do it after finishing the archive just to make sure there can't be a race condition or any other issues I would say. It be fine if it's just one insert at a time I would say (we could always tweak when needed)

Reply:
@tsteur no I mean, in the invalidations table, normally we have all the higher periods before we start archiving, so doing the repair logic here mimics that behavior.

We can still do it after finishing an archive (though there will be less test coverage then).

Reply:
Be great to do it after finishing the archive. Could keep the method and just call it from CronArchive instead of QueueConsumer @diosmosis ? It's really just to avoid random edge cases etc which will happen